### PR TITLE
Enable Y019, Y037, and Y041

### PR DESCRIPTION
**Pull request by twoertwein**
I will try to enable a few more of the flake8-pyi checks in the next days.

### CODE REVIEW

**Comment by Dr-Irv** [comment]
> I will try to enable a few more of the flake8-pyi checks in the next days.

It will be interesting to see if removing `Union` will work for older versions of python.  I think I did a test at some point, and for both `TypeAlias` and `TypeVar` declarations, you had to use `Union` instead of `|` .  But I could be mistaken.  Or maybe you have to use it in runtime code, but removing `Union` in stubs is OK.

**Comment by twoertwein** [comment]
Could also enable Y041. I think we had a discussion about that in the past. Would need to change these cases:

> pandas-stubs/_libs/interval.pyi:151:45: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/series.pyi:1813:30: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/series.pyi:1813:83: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/series.pyi:1814:34: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/series.pyi:1814:87: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/tools/datetimes.pyi:84:19: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)
> pandas-stubs/core/tools/datetimes.pyi:86:13: Y041 Use "float" instead of "int | float" (see "The numeric tower" in PEP 484)

File: pandas-stubs/core/indexes/base.pyi [hunk 1]
```diff
@@ -51,20 +53,18 @@ class InvalidIndexError(Exception): ...
...
+        self,
         idx: slice
         | np_ndarray_anyint
         | Sequence[int]
         | Index
         | Series[bool]
         | Sequence[bool]
         | np_ndarray_bool,
-    ) -> _IndexGetitemMixinT: ...
+    ) -> Self: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/indexes/base.pyi:67]: I thought I tried this here and it didn't work

**Comment by Dr-Irv** [comment]
> Could also enable Y041. I think we had a discussion about that in the past. Would need to change these cases:
> 
> >

I looked at the cases and I would be OK with changing those ones.

The issue we were having before is removing `int` from `Scalar`, and we want to leave it there.

File: pandas-stubs/core/indexes/base.pyi (hunk 1 above)
Comment (twoertwein) [reply pandas-stubs/core/indexes/base.pyi:67]: Locally it works for me (python 3.11.2 with mypy 1.0) but let's see whether it also works on the CI :)

Comment (Dr-Irv) [reply pandas-stubs/core/indexes/base.pyi:67]: seems to have worked

**Review by Dr-Irv — APPROVED** [review]

**Pull request merged**